openapi.yaml

```diff
@@ -9259,3 +9341,149 @@ components:
       - account
       - work_type
       type: object
+    DeploymentMetricID:
+      description: |
+        Supported deployment metrics.
+        * `dep-count` - number of deployments in the time period.
+        * `dep-duration` - average deployment procedure time - the difference between `date_finished` and `date_started` of `DeploymentNotification`.
+        * `dep-success-count` - number of successful deployments in the time period.
+        * `dep-failure-count` - number of failed deployments in the time period.
+        * `dep-success-ratio` - ratio between successful and all deployments.
+        * `dep-size-prs` - average number of deployed pull requests.
+        * `dep-size-releases` - average number of deployed releases.
+        * `dep-size-lines` - average number of deployed lines of code (changed = insertions + removals).
+        * `dep-size-commits` - average number of deployed commits.
+        * `dep-prs-count` - number of pull requests deployed in the time period.
+        * `dep-releases-count` - number of releases deployed in the time period.
+        * `dep-lines-count` - number of deployed lines of code (changed = insertions + removals).
+        * `dep-commits-count` - number of deployed commits.
+        * `dep-repository-frequency` - average number of times a repository was deployed.
```

Comment:
Interesting! What kind of use case do you have in mind for this metric, is it equivalent to `dep-count` using repogroups with groups of 1 repo.

Reply:
It's not equivalent because repogroups filter deployments. Here we analyze the repos deployed in each deployment.
I just thought that:
- It's easy to calculate.
- Good approximation of deployment frequency per repo. Since we don't have `/metrics/repositories`.

Reply:
I'm not sure I understand this correctly. How can this metric give me the release frequencies of the webapp repo and API repo separately for example?